test_natural_movie breaks on recent pandas versions

The index assignment [here](https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/brain_observatory/natural_movie.py#L91) does not like being passed a mock. It complains that:
```
allensdk/brain_observatory/natural_movie.py:74: in sweep_response
    self._sweep_response = self.get_sweep_response()
allensdk/brain_observatory/natural_movie.py:97: in get_sweep_response
    columns=np.array(range(self.numbercells)).astype(str)
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/frame.py:348: in __init__
    mgr = self._init_dict(data, index, columns, dtype=dtype)
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/frame.py:441: in _init_dict
    index = _ensure_index(index)
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/indexes/base.py:4932: in _ensure_index
    return Index(index_like, name=index_like.name, copy=copy)
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/indexes/base.py:435: in __new__
    data, names=name or kwargs.get('names'))
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/indexes/multi.py:1356: in from_tuples
    return MultiIndex.from_arrays(arrays, sortorder=sortorder, names=names)
../../../anaconda3/envs/allensdk-py3/lib/python3.6/site-packages/pandas/core/indexes/multi.py:1305: in from_arrays
    names=names, verify_integrity=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pandas.core.indexes.multi.MultiIndex'>, levels = [], labels = [], sortorder = None, names = <MagicMock name='mock.__getitem__().index.values.name' id='140465540010896'>, dtype = None
copy = False, name = None, verify_integrity = False, _set_identity = True

    def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
                dtype=None, copy=False, name=None,
                verify_integrity=True, _set_identity=True):
    
        # compat with Index
        if name is not None:
            names = name
        if levels is None or labels is None:
            raise TypeError("Must pass both levels and labels")
        if len(levels) != len(labels):
            raise ValueError('Length of levels and labels must be the same.')
        if len(levels) == 0:
>           raise ValueError('Must pass non-zero number of levels/labels')
E           ValueError: Must pass non-zero number of levels/labels

```

--- a/allensdk/test/brain_observatory/test_natural_movie.py
+++ b/allensdk/test/brain_observatory/test_natural_movie.py
@@ -37,10 +37,20 @@ from allensdk.brain_observatory.natural_movie import NaturalMovie
 from allensdk.brain_observatory.stimulus_analysis import StimulusAnalysis
 import pytest
 from mock import patch, MagicMock
+import pandas as pd
 
 
 @pytest.fixture
-def dataset():
+def stimulus_table():
+    return pd.DataFrame([
+        {'frame': 0, 'start': 0, 'stop': 1},
+        {'frame': 0, 'start': 1, 'stop': 2},
+        {'frame': 1, 'start': 2, 'stop': 3},
+    ])
+
+
+@pytest.fixture
+def dataset(stimulus_table):
     dataset = MagicMock(name='dataset')
     
     timestamps = MagicMock(name='timestamps')
@@ -58,7 +68,7 @@ def dataset():
     dataset.get_running_speed=MagicMock(name='get_running_speed',
                                         return_value=(dxcm, dxtime))
     dataset.get_stimulus_table=MagicMock(name='get_stimulus_table',
-                                         return_value=MagicMock())
+                                         return_value=stimulus_table)
     
     return dataset
 

--- a/allensdk/test/brain_observatory/test_natural_movie.py
+++ b/allensdk/test/brain_observatory/test_natural_movie.py
@@ -92,7 +92,12 @@ def mock_sweep_response():
 @patch.object(StimulusAnalysis,
               'get_sweep_response',
               mock_sweep_response())
-@pytest.mark.parametrize('trigger', (1, 2))
+@pytest.mark.parametrize(
+    'trigger', [
+        ('stim_table', 'sweep_response', 'peak'),
+        ('sweeplength', 'sweep_response', 'peak')
+    ]
+)
 def test_harness(dataset, trigger):
     movie_name = "Mock Movie Name"
     nm = NaturalMovie(dataset, movie_name)
@@ -102,14 +107,8 @@ def test_harness(dataset, trigger):
     assert nm._sweep_response is StimulusAnalysis._PRELOAD
     assert nm._peak is StimulusAnalysis._PRELOAD
 
-    if trigger == 1:
-        print(nm.stim_table)
-        print(nm.sweep_response)
-        print(nm.peak)
-    if trigger == 2:
-        print(nm.sweeplength)
-        print(nm.sweep_response)
-        print(nm.peak)
+    for attr in trigger:
+        print(getattr(nm, attr))
 
     assert nm._stim_table is not StimulusAnalysis._PRELOAD
     assert nm._sweeplength is not StimulusAnalysis._PRELOAD

